rlclientlib/azure_factories.cc

```diff
@@ -10,113 +10,135 @@
-    const auto eh_name = cfg.get(name::INTERACTION_EH_NAME, "interaction");
-    const auto eh_url = build_eh_url(eh_host, eh_name);
-    i_http_client* client;
-    RETURN_IF_FAIL(create_http_client(eh_url.c_str(), cfg, &client, status));
-    *retval = new http_transport_client<eventhub_http_authorization>(
-      client,
-      cfg.get_int(name::INTERACTION_EH_TASKS_LIMIT, 16),
-      cfg.get_int(name::INTERACTION_EH_MAX_HTTP_RETRIES, 4),
-      trace_logger,
-      error_cb);
-    return error_code::success;
-  }
+	namespace m = model_management;
+	namespace u = utility;
+
+	int restapi_data_transport_create(m::i_data_transport** retval, const u::configuration& config, i_trace* trace_logger, api_status* status);
+	int episode_sender_create(i_sender** retval, const u::configuration&, error_callback_fn*, i_trace* trace_logger, api_status* status);
+	int observation_sender_create(i_sender** retval, const u::configuration&, error_callback_fn*, i_trace* trace_logger, api_status* status);
+	int interaction_sender_create(i_sender** retval, const u::configuration&, error_callback_fn*, i_trace* trace_logger, api_status* status);
+	int decision_sender_create(i_sender** retval, const u::configuration&, error_callback_fn*, i_trace* trace_logger, api_status* status);
+	int observation_api_sender_create(i_sender** retval, const u::configuration& cfg, error_callback_fn* error_cb, i_trace* trace_logger, api_status* status);
+	int interaction_api_sender_create(i_sender** retval, const u::configuration& cfg, error_callback_fn* error_cb, i_trace* trace_logger, api_status* status);
+
+	void register_azure_factories() {
+		data_transport_factory.register_type(value::AZURE_STORAGE_BLOB, restapi_data_transport_create);
+		sender_factory.register_type(value::OBSERVATION_EH_SENDER, observation_sender_create);
+		sender_factory.register_type(value::INTERACTION_EH_SENDER, interaction_sender_create);
+		sender_factory.register_type(value::EPISODE_EH_SENDER, episode_sender_create);
+		sender_factory.register_type(value::OBSERVATION_HTTP_API_SENDER, observation_api_sender_create);
+		sender_factory.register_type(value::INTERACTION_HTTP_API_SENDER, interaction_api_sender_create);
+	}
+
+	int restapi_data_transport_create(m::i_data_transport** retval, const u::configuration& config, i_trace* trace_logger, api_status* status) {
+		auto uri = config.get(name::MODEL_BLOB_URI, nullptr);
+		auto usingAPI = false;
+
+		if (uri == nullptr) {
+			uri = config.get(name::MODEL_HTTP_APPI_HOST, nullptr);
+			usingAPI = true;
+
+			if (uri == nullptr)
+			{
+				RETURN_ERROR(trace_logger, status, http_model_uri_not_provided);
+			}
+		}
+		i_http_client* client;
+		RETURN_IF_FAIL(create_http_client(uri, config, &client, status));
+
+		if (!usingAPI)
+		{
+			*retval = new m::restapi_data_transport(client, trace_logger);
+		}
+		else
+		{
+			//Initialize header
+			http_headers header;
+			apim_http_authorization* apiObj = new apim_http_authorization();
```

Comment:
it's a public repo so we should not use any internal terminologies like APIM. Bear_http_authroization should be better.

Reply:
renamed the files